lib/src/components/NxToast/NxToast.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useState, useContext, useEffect } from 'react';
+import classnames from 'classnames';
+import FocusContext from './contexts';
+
+import { nxToastPropTypes, NxToastProps, ToastModel } from './types';
+
+export { ToastModel };
+
+const NxToast = (props: NxToastProps) => {
+  const { onClose, children } = props;
+
+  const [isClosing, setIsClosing] = useState(false);
+
+  const focusedToast = useContext(FocusContext);
+
+  useEffect(()=>{
+    focusedToast?.onToastClosing();
+    return () => { Promise.resolve().then(() => focusedToast?.onToastClosing()); };
```

Comment:
Is the promise `then` necessary?

Reply:
> Is the promise `then` necessary?

Promise removed, focusing code is adjusted here: [a10a2d5](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/a10a2d51bd38d9fe3e6b555910ac9271eb3f99cb)